tools/src/bin/rust-indexer.rs

```diff
@@ -349,20 +510,43 @@ fn analyze_file(
         if let Some(ref parent) = parent {
             if parent.kind == DefKind::Trait {
                 let trait_dependent_name = construct_qualname(&parent.qualname, &def.name);
-                visit(
+                visit_def(
                     &mut dataset,
                     AnalysisKind::Def,
                     &def.span,
                     &trait_dependent_name,
                     &def,
                     Some(&parent.qualname),
+                    None,
                 )
             }
         }
 
         let crate_id = &file_analysis.prelude.as_ref().unwrap().crate_id;
         let qualname = crate_independent_qualname(&def, crate_id);
-        visit(&mut dataset, AnalysisKind::Def, &def.span, &qualname, &def, parent.as_ref().map(|p| &*p.qualname))
+        let nested_span = recursive_union_spans_of_def(def, &file_analysis, &defs);
+        let maybe_nested = ignore_boring_spans(&nested_span);
+        visit_def(&mut dataset, AnalysisKind::Def, &def.span, &qualname, &def,
+                  parent.as_ref().map(|p| &*p.qualname), maybe_nested)
+    }
+
+    // We want to expose impls as "def,namespace" with an inferred nesting_range for their
+    // contents.  I don't know if it's a bug or just a dubious design decision, but the impls all
+    // have empty values and no names, so to get a useful string out of them, we need to extract
```

Comment:
Would be good to file an issue against https://github.com/rust-lang/rls for this

Reply:
So the information is actually missing at the save-analysis generation in the compiler, so it's a rust-lang/rust issue.  I filed https://github.com/rust-lang/rust/issues/60872 to track this.